app/controllers/nonsipp/loansmadeoroutstanding/RemoveLoanController.scala

```diff
@@ -110,8 +110,11 @@ class RemoveLoanController @Inject()(
           value =>
             if (value) {
               for {
-                updatedAnswers <- Future
-                  .fromTry(request.userAnswers.remove(IdentityTypePage(srn, index, IdentitySubject.LoanRecipient)))
+                updatedAnswers <- Future.fromTry(
+                  request.userAnswers
+                    .remove(IdentityTypePage(srn, index, IdentitySubject.LoanRecipient))
+                    .remove(LoansProgress(srn, index))
```

Comment:
If it makes it simpler, keep this here, but it really should be in cleanup action of IdentityTypePage (conditionally on IdentitySubject.LoanRecipient)

Reply:
This will be addressed